src/public/modules/forms/services/submissions.client.factory.js

```diff
@@ -93,23 +93,23 @@ function SubmissionsFactory(
     xhr.send(fd)
     // On Response
     xhr.onreadystatechange = function () {
-      // 4 DONE  The operation is complete.
       const OPERATION_DONE = 4
       if (xhr.readyState === OPERATION_DONE) {
         // waterfall is successful
-        if (xhr.status === HttpStatus.OK) {
-          deferred.resolve('Submission has finished.')
-        } else {
-          let response = {}
-          try {
-            response = JSON.parse(xhr.responseText)
-            // eslint-disable-next-line no-empty
-          } catch (e) {}
+        let response = {}
+        try {
+          response = JSON.parse(xhr.responseText)
+          if (xhr.status === HttpStatus.OK) {
+            deferred.resolve({
+              message: 'Submission has finished.',
+              submissionId: response.submissionId,
+            })
+          } else {
+            deferred.reject(`${response.message}`)
```

Comment:
let's keep the default message here as a backup, i.e. `deferred.reject(response.message || 'Please refresh...`